lib/offline_sort/offline_sort.rb

```diff
@@ -20,12 +21,31 @@ def initialize(enumerable, chunk_input_output_class: DEFAULT_CHUNK_IO_CLASS, chu
     end
 
     def sort
-      merge(split)
+      sorted_chunks = []
+      chunk_entries = []
+
+      enumerable.each do |entry|
+        chunk_entries << entry
+
+        if chunk_entries.size == chunk_size
+          sorted_chunks << write_sorted_chunk(chunk_entries)
+          chunk_entries.clear
+        end
+      end
+
+      unless chunk_entries.empty?
```

Comment:
I know it was in the old `split` method, but I don't totally understand this case. Is it for when the input is an empty enumerable? If that's the case can we just `return [] if chunk_entries.empty?` instead of going through the `merge` method too? Or are there cases where it might not be an array?

Reply:
It's for when there's a non-full chunk, which would usually happen on the last few entries. For example, if you had a chunk size of 10 say, and 25 entries, the first two chunks (20 entries) would be written into `sorted_chunks` on line 31. But we'd still have 5 entries in chunk_entries. We'd never get to the case in the if on line 30, but we would still want to write out these final entries (and, importantly, save them to `sorted_chunks`). So we do this final check here for any leftover chunks. Does that make sense?

Reply:
Ah yes, of course! I saw that earlier and it made sense and then right when I was approving I forgot all of that context. :lgtm:!